### PR TITLE
Bugfix: `anyOf` and `array` schema mapping

### PR DESCRIPTION
**Pull request by znicholasbrown**
This PR makes two important changes to the way array values are mapped in schemas:

1. The property array response mapper first checks if a value is already a JSON string before attempting to stringify it for use in code input components
2. Schemas attempt to parse a value when deciding which union property to display

There's still a bug here where validators for fieldsets in the union will apply to the incorrect member, but I'm working on that

### CODE REVIEW

File: src/services/schemas/utilities.ts [hunk 1]
```diff
@@ -231,15 +231,23 @@ export function getSchemaValueDefinitionIndex(definitions: Schema[], value: Sche
...
   }
 
-  switch (typeof value) {
+  let parsedValue = value
+
+  try {
+    parsedValue = parseUnknownJson(value)
+  } catch {
+    // do nothing, use the raw value
+  }
```
Comment (pleek91) [inline src/services/schemas/utilities.ts:240]: Isn't this going to mess up some types that could be parsed as json but are actually just supposed to be strings?

File: src/services/schemas/properties/SchemaPropertyArray.ts [hunk 2]
```diff
@@ -35,6 +35,10 @@ export class SchemaPropertyArray extends SchemaPropertyService {
 
   protected response(value: SchemaValue): unknown {
     if (this.componentIs(JsonInput)) {
+      if (typeof value === 'string') {
+        return value
+      }
```
[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/services/schemas/properties/SchemaPropertyArray.ts:40]: In the response? I don't _think_ we need to. So long as it's a string, the validators at the component level should catch that and the json input components don't really care if it's valid

File: src/services/schemas/utilities.ts (hunk 1 above)
Comment (znicholasbrown) [reply src/services/schemas/utilities.ts:240]: For posterity: after chatting we've decided to add a check on the `parsedValue`'s type as well and only continue with objects so as not to unintentionally parse strings that contain numbers, booleans, etc

File: src/services/schemas/utilities.ts [hunk 3]
```diff
@@ -231,6 +232,16 @@ export function getSchemaValueDefinitionIndex(definitions: Schema[], value: Sche
...
 
+  try {
+    const parsedValue = parseUnknownJson(value)
+
+    if (isRecord(parsedValue) || Array.isArray(parsedValue)) {
+      return findObjectDefinitionIndex(definitions, parsedValue)
+    }
+  } catch {
+    // do nothing, use the raw value
+  }
```
Comment (pleek91) [inline src/services/schemas/utilities.ts:243]: We don't need a try catch here do we? `parseUnknownJson()` already silences the error.

Comment (znicholasbrown) [reply src/services/schemas/utilities.ts]: Ah good point - updated

**Comment by znicholasbrown** [comment]
Also found the source of the non-reactive field-level validators @pleek91 so those are fixed in this as well

**Review by pleek91 — APPROVED** [review]

**Pull request merged**